pages/api/applications/[applicationId].ts

```diff
@@ -1,5 +1,112 @@
+import { PrismaClient } from '@prisma/client';
 import type { NextApiRequest, NextApiResponse } from 'next';
+import { ApiResponse, EmptyPayload, StatusMessage, StatusMessageType } from '../../../types/apiResponse';
+import { ApplicationData } from '../../../types/application';
+import { withVerifiedUser } from '../../../utils/auth/jwtHelpers';
+import {
+  createJsonResponse,
+  HTTP_DELETE_METHOD,
+  HTTP_GET_METHOD,
+  HTTP_STATUS_NOT_FOUND,
+  HTTP_STATUS_OK,
+  HTTP_STATUS_UNAUTHORIZED,
+  rejectHttpMethod,
+} from '../../../utils/http/httpHelper';
 
-export default function handler(req: NextApiRequest, res: NextApiResponse) {
-  res.status(200).end(`Request sent to ${req.url}\n`);
+const prisma = new PrismaClient();
+
+enum MessageType {
+  APPLICATION_NOT_FOUND,
+  APPLICATION_DOES_NOT_BELONG_TO_USER,
+  APPLICATION_DELETED_SUCCESSFULLY,
+}
+
+const messages = new Map<MessageType, StatusMessage[]>([
+  [MessageType.APPLICATION_NOT_FOUND, [{ type: StatusMessageType.Error, message: 'Application cannot be found.' }]],
+  [
+    MessageType.APPLICATION_DOES_NOT_BELONG_TO_USER,
+    [{ type: StatusMessageType.Error, message: 'Application does not belong to you.' }],
+  ],
+  [
+    MessageType.APPLICATION_DELETED_SUCCESSFULLY,
+    [{ type: StatusMessageType.Success, message: 'Application was deleted successfully.' }],
+  ],
+]);
+
+function handler(userId: string, req: NextApiRequest, res: NextApiResponse) {
+  const method = req.method;
+
+  switch (method) {
+    case HTTP_GET_METHOD:
+      handleGet(userId, req, res);
+      break;
+    case HTTP_DELETE_METHOD:
+      handleDelete(userId, req, res);
+      break;
+    default:
+      rejectHttpMethod(res, method);
+  }
+}
+
+async function handleGet(
+  userId: string,
+  req: NextApiRequest,
+  res: NextApiResponse<ApiResponse<ApplicationData | EmptyPayload>>,
+) {
+  const applicationId = Number(req.query.applicationId);
```

Comment:
We wouldn't hit this route if its not a number right?

Reply:
It can, e.g. `api/applications/foo` endpoint. We'll then get a prism client validation error:

error - unhandledRejection: PrismaClientValidationError: 
Invalid `prisma.application.findFirst()` invocation:

{
  where: {
    id: NaN,
        ~~~

Reply:
@bharathcs I still don't get it. The application id is from the url query params. If it wasn't a number why would this function run?

Reply:
Query parameters can be strings like "foo" right? Or is this some REST thing I'm not aware of?

Reply:
@bharathcs you are right. will fix it.